pkg/overlay/cache_test.go

```diff
@@ -81,17 +81,16 @@ func bootstrapTestNetwork(t *testing.T, ip, port string) ([]dht.DHT, pb.Node) {
 	dhts := []dht.DHT{}
 
 	p, err := strconv.Atoi(port)
-	pm := strconv.Itoa(p)
 	assert.NoError(t, err)
-	intro, err := kademlia.GetIntroNode(fmt.Sprintf("%s:%s", ip, pm))
+	intro, err := kademlia.GetIntroNode(net.JoinHostPort(ip, port))
```

Comment:
This wouldn't work since `127.0.0.1:0`, would only be defined after starting the server.